lib/bloc/search/search_event.dart

```diff
@@ -7,9 +8,14 @@ abstract class SearchEvent {
 }
 
 class CourseTapEvent extends SearchEvent {
-  CourseTapEvent(String label) : super(label);
+  CourseTapEvent(String label) : super(label){
+    LocalKeyValuePersistence.updateSearchHistory(label);
```

Comment:
I believe this bloc event gets dispatched regardless of which instance of the search fragment it's called on.

Based on the way that the local storage is structured, this means that everything will get added to the same list.

Should we make it so that each list has a unique id in order to support saving multiple search histories?

Reply:
Do we need search somewhere else for now? Maybe is okay to share the same search history across the app. I am not sure.